### DISCUSSION TITLE
Registering a custom URI scheme example

### DISCUSSION THREAD

**Discussion by errmayank** [tauri-apps/tauri#5597] [Q&A] [answered]
Is there an example that shows how to set up custom URI scheme for a tauri app?

I see [here](https://docs.rs/tauri/latest/tauri/struct.Builder.html#method.register_uri_scheme_protocol) that we need to pass in a callback function that takes a URL and returns `Result` but i can't figure it out, an example would be really helpful!
```rs
tauri::Builder::default().register_uri_scheme_protocol("example", ?)
  .run(tauri::generate_context!())
  .expect("error while running tauri application")
```

**Comment by FabianLars** [comment, marked as answer]
Take a look at these example:
https://github.com/tauri-apps/tauri/blob/dev/examples/streaming/main.rs
https://github.com/JonasKruckenberg/pisano/blob/main/src-tauri/src/main.rs#L43

**Reply by errmayank** [reply]
@FabianLars thanks but i tried both and couldn't get it to work still

I went ahead and installed pisano app but seems like even that is not running on a custom protocol?
or am i missing something?

<img width="926" alt="Screenshot 2022-11-12 at 15 18 43" src="https://user-images.githubusercontent.com/64462998/201468722-da179995-9268-48dd-9ce6-6391f43604db.png">

What i'm looking for is something what slack does by going to `slack://open`

<img width="986" alt="Screenshot 2022-11-12 at 15 19 36" src="https://user-images.githubusercontent.com/64462998/201468819-323615c9-c0fd-4703-bffa-46b4d52a3b69.png">

**Reply by FabianLars** [reply]
Ahh yeah that's something completely different, sorry about that. What you're looking for is #323 (not implemented yet, can be implemented with a bit of rust or wix fragments yourself if you only care about windows) - please don't add a `+1` there 🙏 

But tbh i'm still a bit confused because this: "I went ahead and installed pisano app but seems like even that is not running on a custom protocol?" and this "What i'm looking for is something what slack does by going to slack://open" are really 2 different things.
The first thing sounds like you're asking about an alternative to `tauri://localhost` (which is an alternative scheme to http which only works in the tauri app) and the second one is a "deep link" used to open other apps and communicate with them (these apps will still run on the `tauri://` scheme in the context of tauri)

**Reply by errmayank** [reply]
Oh, thanks for clarifying! i thought these two were the same thing 😅
